cmd/exporter/exporter.go

```diff
@@ -150,7 +148,7 @@ func main() {
 		klog.Fatalf("failed to create eBPF exporter: %v", err)
 	}
 	defer bpfExporter.Detach()
-
+	stats.InitAvailableParamAndMetrics(bpfExporter.SupportedMetrics())
```

Comment:
@maryamtahhan this is very similar to how it used to be before https://github.com/sustainable-computing-io/kepler/commit/0baec479033aaca3d1744ec3010e156afd0075e8

The reason I fixed this by passing it around everywhere was that there is still a risk of racing here.

i.e what is going to prevent the following scenarios:

1. stats.NewStats (or one of the variants) being called **before** `InitAvailableParamAndMetrics`
1. Access to `stats.AvailableBPFMetrics` **before**  `InitAvailableParamAndMetrics`

Reply:
I'm obviously missing something (sorry); I'm not sure how passing that bpfsupportedmetrics around fixed the risk of racing, when I tracked down the usage of bpfsupportedmetrics the majority of places led to NewStats(), so we were just passing this variable along through n levels of functions so that n stats has a reference to the bpfmetrics. 

From what I can see the model isn't initialised till the Collector is started and in order to create the collector the bpfexporter needs to be created and part of that is that the bpf stats are initialised. we were making a call to `stats.InitAvailableParamAndMetrics(bpfExporter.SupportedMetrics())` directly after bpfexporter creation.

Regardless I will work on this patch to make it so that stats will no longer depend on the bpf package and revert the model to use the bpf package as it did originally and push another patch.

Reply:
I don't disagree with your assessment of how it works right now 😄 
The problem is that we're encoding assumptions into the codebase - this will happen, before that.
These are assumptions that no compiler or linter can help ensure remain true, especially as these assumptions seem to cross package/module boundaries.

I'm all for letting the compiler do the hard work. So if thing X needs and instance of thing Y to operate, it has to be given one when it's constructed.